pretyping/pretyping.ml

```diff
@@ -517,6 +517,15 @@ let rec pretype k0 resolve_tc (tycon : type_constraint) (env : GlobEnv.t) (sigma
         | Some ty -> sigma, ty
         | None -> new_type_evar env sigma loc in
       let sigma, uj_val = new_evar env sigma ~src:(loc,k) ~naming ty in
+      let sigma =
```

Comment:
How is this change related to this PR?

Reply:
It moves the handling of the obligation flag higher in the hierarchy, avoiding to do strange comparisons of the evar kind in unification.